File: AzureStorageSDK/Java/DownloadAzureStorage.java
import java.io.FileOutputStream;

import com.microsoft.azure.storage.*;
import com.microsoft.azure.storage.blob.*;

public class DownloadAzureStorage {
	static String mycontainer = "images";
	static String mypath = "c:\\demo\\AzureStorageSDK\\images\\";

	static String storageConnectionString = "DefaultEndpointsProtocol=https;" +
		    "AccountName=dev018storage;" +
		    "AccountKey=q4dLJhW3zLw8hmdjL88w0vc2KHrbjMJlXumFB0eZVpBoeks1MKlZh7+pW36fmlWmBMx+bzwvWcYMU8p4MGhbcg==";

	public static void main(String[] args) {
		try
		{
		    // Retrieve storage account from connection-string.
		    CloudStorageAccount storageAccount = CloudStorageAccount.parse(storageConnectionString);

		    // Create the blob client.
		    CloudBlobClient blobClient = storageAccount.createCloudBlobClient();

		    // Retrieve reference to a previously created container.
		    CloudBlobContainer container = blobClient.getContainerReference(mycontainer);
		    System.out.println("\nDownload by Java\n");
		    // Loop over blobs within the container and output the URI to each of them.
		    for (ListBlobItem blobItem : container.listBlobs()) {
		    	 // If the item is a blob, not a virtual directory.
		        if (blobItem instanceof CloudBlob) {
		            // Download the item and save it to a file with the same name.
		             CloudBlob blob = (CloudBlob) blobItem;
		             blob.download(new FileOutputStream(mypath + blob.getName()));
		             System.out.println("\""+blob.getName()+"\" Downloaded!\n");
		             // Delete the blob.
		             blob.deleteIfExists();
		        }
		    }
		}
		catch (Exception e)
		{
		    // Output the stack trace.
		    e.printStackTrace();
		}
	}
}

File: AzureStorageSDK/Python/DownloadAzureStorage.py
from azure.storage import BlobService 

mycontainer = 'images'
mypath = 'c:\\demo\\AzureStorageSDK\\images\\'

blob_service = BlobService(
    account_name='dev018storage', 
    account_key='q4dLJhW3zLw8hmdjL88w0vc2KHrbjMJlXumFB0eZVpBoeks1MKlZh7+pW36fmlWmBMx+bzwvWcYMU8p4MGhbcg=='
)

blobs = blob_service.list_blobs(mycontainer)
print "\nDownload by Python\n"
for blob in blobs:
    blob_service.get_blob_to_path(mycontainer, blob.name, mypath+blob.name)
    print("\""+blob.name+"\" Downloaded!\n")
    blob_service.delete_blob(mycontainer, blob.name) 